src/components/Buttons/VoteOnButton.jsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { voteOnArticle } from "../../utils/api";
+
+const VoteOnButton = ({ setVotes }) => {
+  const { articleId } = useParams();
+  const [voteMsg, setVoteMsg] = useState({ inc_votes: 0 });
+  const [err, setErr] = useState(null);
+
+  const updateVotes = (vote) => {
+    setVotes((currVotes) => currVotes + vote);
+    setVoteMsg((currVotesMsg) => ({ ...currVotesMsg, inc_votes: vote }));
+  };
+
+  useEffect(() => {
+    voteOnArticle(articleId, voteMsg).catch((err) => {
+      console.log(err.response.status, err.response.data.msg);
+      setErr(err.response);
+      if (voteMsg.inc_votes === 1) {
+        setVotes((currVotes) => currVotes - 1);
+      } else {
+        setVotes((currVotes) => currVotes + 1);
+      }
+    });
+  }, [voteMsg]);
```

Comment:
Im not sure why this is in a useEffect - do you what this to happen when the component mounts? 

I presume it sends an api request of zero when the component mounts, then again later. You've got extra api calls happening. 

You can just create a function and invoke it yourself. - you dont need a use Effect for this